src/ipa_pc/data_structures.rs

```diff
@@ -892,3 +927,463 @@ impl<F: PrimeField> CanonicalDeserialize for SuccinctCheckPolynomial<F> {
         Ok(SuccinctCheckPolynomial(values))
     }
 }
+
+/// 'Segmentized' commitment to a polynomial that optionally enforces a degree bound.
+#[derive(Derivative)]
+#[derivative(
+Default(bound = ""),
+Hash(bound = ""),
+Clone(bound = ""),
+Debug(bound = ""),
+PartialEq(bound = ""),
+Eq(bound = "")
+)]
+pub struct SegmentedCommitment<G: AffineCurve> {
+    /// As we use segmentation, a commitment consists of several single-segment commitments.
+    pub comm: Vec<G>,
+}
+
+impl<G: AffineCurve> LinearCombination<G::ScalarField> for SegmentedCommitment<G> {
+    type Item = SegmentedCommitment<G>;
+
+    fn from_lc(_items: Vec<Self::Item>, _base: G::ScalarField, _step: usize) -> Self::Item {
+        Self::Item::empty()
+    }
+}
```

Comment:
Actually, the idea was to define `pub struct ExtensionGroup<G:Group>` as vectors of elements from `G`,  implement the group trait for `ExtensionGroup`, and then `PolynomialCommitment<ExtensionGroup<G>>` for the `DomainExtendedPolynomialCommitment<G>`. Hence the entire domain extension is generic, as it should be in my opinion: the only property one needs to extend is linearity of the commitment scheme.